app/+QSP/@Settings/refreshData.m

```diff
@@ -1,110 +1,134 @@
-function [StatusOk,Message] = refreshData(obj)
-
-StatusOk = true;
-Message = '';
-
-hWaitBar = uix.utility.CustomWaitbar(0, 'Loading session', '', true);
-
-
-%% Task
-
-for index = 1:numel(obj.Task)
-    ProjectPath = obj.Task(index).FilePath;
-    ModelName = obj.Task(index).ModelName;
-    MaxWallClockTime = obj.Task(index).MaxWallClockTime;
-    [ThisStatusOk,ThisMessage] = importModel(obj.Task(index),ProjectPath,ModelName);
-    obj.Task(index).MaxWallClockTime = MaxWallClockTime;
-    if ~ThisStatusOk
-        StatusOk = false;
-        Message = sprintf('%s\n%s\n',Message,ThisMessage);
+function [StatusOk,Message] = refreshData(obj, uiTF)
+    % TODO: This should be a method of Settings. [REFACTOR]
+    arguments
+        obj(1,1) QSP.Settings
+        uiTF(1,1) logical = false
     end
-    waitStatus = uix.utility.CustomWaitbar(index/numel(obj.Task), hWaitBar, 'Loading Tasks');
-    if ~waitStatus % cancelled
-        StatusOk = false;
-        obj = QSP.Settings.empty;
-        delete(hWaitBar)
-        error('Settings:CancelledLoad', 'Cancelled load');
+    
+    if uiTF
+        waitbar = @(arg1, arg2, arg3, arg4)uix.utility.CustomWaitbar(arg1, arg2, arg3, arg4);
+    else
+        waitbar = @(arg1, arg2, arg3, arg4)true;
     end
-end
-
-if nargout==3
-    varargout{1} = false;
-end
-
-
-%% Virtual Population
-
-for index = 1:numel(obj.VirtualPopulation)
-    FilePath = obj.VirtualPopulation(index).FilePath;
-    [ThisStatusOk,ThisMessage] = importData(obj.VirtualPopulation(index),FilePath);
-    if ~ThisStatusOk
-        StatusOk = false;
-        Message = sprintf('%s\n%s\n',Message,ThisMessage);
+    
+    StatusOk = true;
+    Message = '';
+    
+    hWaitBar = waitbar(0, 'Loading session', '', true);    
```

Comment:
seems like this will work for now. in the future let's move all the waitbars into the viewer?

Reply:
yes, all UI elements (including waitbars should be in the viewer).